client/belga/image/belga-coverage-info.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+
+import {IBelgaCoverage, getCoverageInfo} from './belga-image-api';
+
+interface IProps {
+    coverageId: string;
+    removeCoverage?: () => void;
+}
+
+interface IState {
+    loading: boolean;
+    coverage: IBelgaCoverage;
+}
+
+export default class BelgaCoverageAssocation extends React.Component<IProps, IState> {
+    readonly state = {loading: true, coverage: null};
+
+    componentDidMount() {
+        this.fetchCoverage();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.coverageId !== this.props.coverageId) {
+            if (this.props.coverageId) {
+                this.setState({loading: true, coverage: null});
+                this.fetchCoverage();
+            } else {
+                this.setState({loading: false, coverage: null})
+            }
+        }
+    }
+
+    fetchCoverage() {
+        getCoverageInfo(this.props.coverageId)
+            .then((coverage) => this.setState({coverage: coverage, loading: false}))
+            .catch(() => {
+                this.setState({loading: false});
+            });
+    }
+
+    render() {
+        if (this.state.loading) {
+            return null;
+        }
+
+        if (this.state.coverage != null) {
+            return this.renderCoverage(this.state.coverage);
+        }
+
+        return <div>{'There was an error when fetching coverage.'}</div>;
+    }
+
+    renderCoverage(coverage: IBelgaCoverage) {
+        const inEditor = this.props.removeCoverage != null;
+        return (
+            <div className={inEditor ? "sd-media-carousel__content" : ''}>
```

Comment:
I came up with an idea to solve CSS interdependencies. Belga coverages shouldn't depend on CSS used by other features in client-core. How about if we create a React UI component, say Carousel, export it from the extensions API, use it here like

```
<Carousel>
    <CarouselItem caption={'test'}>
        <img />
    </CarouselItem>
    <CarouselItem>
        <img />
    </CarouselItem>
</Carousel>
```

The implementation would still use CSS classes related to media carousel/associations, but we could then hand this over to the UI team to refactor the CSS and move it to UI framework when it's done. What do you think?

Reply:
yep that would be ideal, will add some core components for these